cirq/ion/ion_gates.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Operations native to iontrap systems."""
+from typing import (
+    Union, Tuple, Optional, List, Callable, cast, Iterable, Sequence,
+)
+
+import numpy as np
+
+from cirq import value, linalg, protocols
+from cirq.ops import gate_features, eigen_gate, raw_types, gate_operation
+from cirq.type_workarounds import NotImplementedType
+
+
+
+class MSGate(eigen_gate.EigenGate,
```

Comment:
This shouldn't be a separate class, it should be the following method (plus docstring):

```python
def MS(rads: float) -> XXPowGate:
    return ops.XXPowGate(exponent=rads*np.pi, global_shift=-0.5)
```